app/javascript/components/EventsTimeline.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, Fragment } from "react";
+import Select, {components} from 'react-select';
+import PropTypes from "prop-types";
+import Testimonials from "./Testimonials";
+import moment from 'moment';
+
+Testimonials.propTypes = {
+  events: PropTypes.array,
+  options: PropTypes.array,
+};
+
+const blueDarkColor = '#2E3152';
+
+const customStyles = {
+  option: (provided, state) => ({
+    ...provided,
+    color: blueDarkColor,
+    backgroundColor: state.isSelected ? 'inherit' : 'inherit',
+  }),
+  container: (provided) => ({
+    ...provided,
+    display: 'inline-block',
+  }),
+  control: (provided, state) => ({
+    ...provided,
+    borderRadius: 0,
+    borderColor: state.isFocused ? blueDarkColor : blueDarkColor,
+    minWidth: 200,
+  }),
+  menu: (provided) => ({
+    ...provided,
+    borderRadius: 0,
+    marginTop: 0,
+    width: 250,
+    right: 0,
+    zIndex: 9,
+  }),
+}
+
+const Option = props => (
+  <div>
+    <components.Option {...props}>
+      <input type="checkbox" hidden checked={props.isSelected} onChange={() => null} />
+      {props.isSelected && <div className="checked select-checkbox"><i className="fa fa-check"></i></div>}
+      {!props.isSelected && <div className="unchecked select-checkbox"></div>}
+      <label>{props.label}</label>
+    </components.Option>
+  </div>
+);
+
+const EventsTimeline = ({ events, options, isFiltered = false }) => {
+  const [currentTypes, setCurrentTypes] = useState(false);
+  const el = React.createRef();
```

Comment:
A better name here would be nice. Took a bit to figure out which element el was.